src/HDF5.jl

```diff
@@ -1969,7 +1982,7 @@ end
 ### Convenience wrappers ###
 # These supply default values where possible
 # See also the "special handling" section below
-h5a_write(attr_id::Hid, mem_type_id::Hid, buf::String) = h5a_write(attr_id, mem_type_id, Vector{UInt8}(buf))
+h5a_write(attr_id::Hid, mem_type_id::Hid, buf::String) = h5a_write(attr_id, mem_type_id, Vector{UInt8}(codeunits(buf)))
```

Comment:
isn't it better to use `unsafe_wrap` here instead to avoid the extra allocation?

Reply:
Done.